bot/common/threads/thread_builder.py

```diff
@@ -385,7 +387,10 @@ class Step:
     current: BaseStep
     next_steps: Optional[Dict[str, BaseStep]] = field(default_factory=dict)
     previous_step: Optional[BaseStep] = field(default=None)
-    hash_: str = hashlib.sha256("".encode()).hexdigest()
+    hash_: str = field(default=None)
+
+    def __post_init__(self):
+        self.hash_ = hashlib.sha256(self.current.name.encode()).hexdigest()
```

Comment:
Should name be a required property on baseStep? It might also be good to add like a custom string because I think this doesn't solve the case of a fork with two steps that are the same, but have different arguments.

Reply:
It likely should be required explicitly, since it's assumed everywhere in code but largely unstated. Also good point about that case with the arguments. let's tackle that one when we need to add two steps of the same type (but different args) to the same fork?

Reply:
Yes, let's create a ticket for it then and add it as tech debt. I can totally see us forgetting about this lol